packages/medusa/src/interfaces/csv-parser.ts

```diff
@@ -0,0 +1,34 @@
+import { AwilixContainer } from "awilix"
+import { IParser } from "./abstract-parser"
+
+export type CsvParserContext = {
+  line: number
+  column: string
+}
+
+export interface IValidator<TLine> {
+  validate: (
+    value: string,
+    line: TLine,
+    context: CsvParserContext
+  ) => Promise<boolean>
+}
+
+export abstract class AbstractValidator<TLine> implements IValidator<TLine> {
+  constructor(protected readonly container: AwilixContainer) {}
+
+  abstract validate(value: string, line: TLine, context: CsvParserContext)
+}
+
+export type CsvSchema<TLine = any> = {
```

Comment:
Indeed the any could be removed 👍

Reply:
i guess not, but when removing the `any`, we'll have to add another generic so that `CsvSchema` works [here](https://github.com/medusajs/medusa/blob/1189291e619da19a7a1dc308afd8d3b813752211/packages/medusa/src/services/csv-parser.ts#L12), not much wrong with that ofc, but feels like a bit too much 😅 wdyt?